rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,538 @@
+- `nextTripKey` (Trip Key, optional): links the newly created trip to the trip after it. If the Trip Key refers to an added trip, it may not have been seen in the event stream at the time of this message
+- `consist` (array of Optional Car, optional, default: empty list): the cars assigned to perform this trip. If a car is `null`, then no car is currently assigned to that position in the consist.
+- `operators` (list of Optional Operator, optional, default: empty list): the array of operators assigned to perform this trip. If an operator is `null`, then no operator is assigned to the car in that position in the consist.
+- `comment` (string, optional): free text with additional information about the trip
+
+*Notes*
+- The event should only include values which were explicitly included by the `author`: inferred values should not be included.
+- `consist` and `operators` MUST be the same length; if they are not, the behavior of consumers is undefined.
+
+### Trip arrived
+Inspectors are required to note when trains arrive at their location. This event is generated by that confirmation, and includes the relevant information about who performed the trip. 
+
+Event type: `com.mbta.ctd.glides.trip-arrived`
+Fields in the event:
+- `author` (Author): the inspector adding the trip.
+- `tripKey` (Trip Key): which trip is being confirmed.
+- `comment` (string, optional): free text information about the trip.
+- `location` (Location): the location at which the train has arrived.
+- `arrivalTime` (Time): if present, the time that the train was recorded arriving at the specified location.
+- `consist` (array of Car): the cars which performed this trip.
+- `operators` (array of Operator): the list of operators who to performed this trip.Even if the scheduled operators performed the trip, those values MUST be included here.
+
+*Notes*
+- `consist` and `operators` MUST be the same length; if they are not, the behavior of consumers is undefined.
+
+### Trip departed
+Inspectors are required to note when trains arrive at their location. This event is generated by that confirmation, and includes the relevant information about who performed the trip. 
+
+Event type: `com.mbta.ctd.glides.trip-departed`
+Fields in the event:
+- `author` (Author): the inspector adding the trip
+- `tripKey` (Trip Key): which trip is being confirmed
+- `comment` (string, optional): free text information about the trip
+- `location` (Location): the location from which the train has departed.
+- `departureTime` (Time): the time that the train was recorded departing from the specified location.
+- `consist` (array of Car): the cars which performed this trip.
+- `operators` (array of Operator): the list of operators who to performed this trip. Even if the scheduled operators performed the trip, those values MUST be included here.
```

Comment:
Ah yes, this addresses the issue of the consumer knowing the schedule, and giving the operators when the trip is departing is a good time for that.